### PR TITLE
Add OffsetComponents trait

### PR DESCRIPTION
**Pull request by ianthetechie**
This covers part of the additions discussed in #54.

### CODE REVIEW

**Comment by quodlibetor** [comment]
I'm not sure why tests aren't running, it's possible that applying this commit will make them run for you: https://github.com/chronotope/chrono-tz/commit/f61e396582feaec2198c9710db4e6d3f1ab3388c

**Comment by ianthetechie** [comment]
Looks like it's running now after pulling in that commit.

**Comment by quodlibetor** [comment]
I'm probably a little busy until Friday and I want to take a deeper look but at a first glance that this looks great!

**Comment by ianthetechie** [comment]
No worries; take your time! As there isn't much precedent for extra traits like this in this crate I'm sure you'll want to make sure the organization etc. are correct.

File: src/lib.rs [hunk 1]
```diff
@@ -82,6 +82,28 @@
...
+//!
+//! # fn main() {
+//! let london_time = London.ymd(2016, 5, 10).and_hms(12, 0, 0);
+//!
+//! // London typically has zero offset from UTC, but has a 1h adjustment forward
+//! // when summer time is in effect.
+//! assert_eq!(london_time.offset().base_utc_offset(), Duration::hours(0));
+//! assert_eq!(london_time.offset().dst_offset(), Duration::hours(1));
+//! # }
+//! ```
```
[the remark this answers is not in the document]
Comment (ianthetechie) [reply src/lib.rs]: Good call. Updated.

**Comment by quodlibetor** [comment]
Thanks for this!

**Pull request merged**